pkg/pipeline/stream/destination_acker.go

```diff
@@ -131,22 +131,22 @@ func (n *DestinationAckerNode) worker(
 			msg := n.queue.PopFront()
 			n.m.Unlock()
 
-			pos, err := n.Destination.Ack(ctx)
-			if pos == nil {
-				// empty position is returned only if an actual error happened
-				handleError(msg, cerrors.Errorf("failed to receive ack: %w", err))
-				return
-			}
-			if !bytes.Equal(msg.Record.Position, pos) {
-				handleError(msg, cerrors.Errorf("received unexpected ack, expected position %q but got %q", msg.Record.Position, pos))
-				return
-			}
-
-			err = n.handleAck(msg, err)
+			acks, err := n.Destination.Ack(ctx)
 			if err != nil {
-				errChan <- err
+				handleError(msg, cerrors.Errorf("error while fetching acks: %w", err))
 				return
 			}
+			for _, ack := range acks {
```

Comment:
Only 1 ack is expected here I guess?

Reply:
For now, this will be a single ack, yes. But this specific code should already work with multiple acks.